_posts/2024-12-xx-hybrid-search-optimization.md

```diff
@@ -0,0 +1,342 @@
+
+| Metric | Baseline BM25 – Small | Global Hybrid Search Optimizer – Small | Baseline BM25 – Large | Global Hybrid Search Optimizer – Large |
+| :---: | :---: | :---: | :---: | :---: |
+| DCG@10 | 9.65 | 9.99 | 8.82 | 9.30 |
+| NDCG@10 | 0.24 | 0.26 | 0.23 | 0.25 |
+| Precision@10 | 0.27 | 0.29 | 0.24 | 0.27 |
+
+Improvements are seen across all metrics for both datasets. To recap, up to this point, we performed the following steps:
+
+* Create a query set by randomly sampling.
+* Generate judgments (to be precise, we only used the existing judgments of the ESCI dataset).
+* Calculate search metrics for a baseline.
+* Try out several hybrid search combinations.
+* Compare search metrics.
+
+Two things are important to note:
+
+* While the systematic approach can be transferred to other applications, the experiment results cannot. It is necessary to always evaluate and experiment with your own data.  
+* The ESCI dataset does not provide 100% judgment coverage. On average we saw roughly 35% judgment coverage among the top 10 retrieved results per query. This leaves us with some uncertainty.
+
+The improvements tell us that we optimize our metrics on average when switching to hybrid search with the above parameter values. But of course there are queries that benefit (as in their search quality metrics improve) and queries that do not benefit (as in their search quality metrics decrease) when conducting this switch. This is something we can virtually always observe when comparing two search configurations with each other. While one configuration outperforms the other on average, not every query will profit from the configuration.
+
+The following chart shows the DCG@10 values of the training queries of the small query set. The x-axis represents the search pipeline with l2 norm, arithmetic mean, 0.1 lexical search weight, and 0.9 neural search weight (configuration A). The y-axis represents the search pipeline with an identical normalization and combination technique but switched weights: 0.9 lexical search weight and 0.1 neural search weight (configuration B).
+
+<img src="/assets/media/blog-images/2024-12-xx-optimizing-hybrid-search/1_search_config_comparison.png" alt="Scatter Plot of DCG values for lexical-heavy search configuration and Neural-heavy search configuration" class="center"/>{:style="width: 100%; max-width: 800px; height: auto; text-align: center"}
+
+The queries with the highest search quality metrics improvements of configuration B are those that are located on the y-axis: they have a DCG score of 0 for this configuration. And for configuration A some even score above 15.
+
+As we strive for improving the search quality metrics for all queries this now leads us to the question: improvements on average are fine but how can we tackle this in a more targeted way to come up with an approach that provides us the best configuration per-query instead of one good configuration for all queries?  
```

Comment:
```suggestion
Striving to improve the search quality metrics for all queries raises the following question: improvements on average are fine, but how can we tackle this in a more targeted way to come up with an approach that provides the best configuration per query instead of one good configuration for all queries?  
```